src/providers/generalWalletProvider.tsx

```diff
@@ -343,6 +347,9 @@ export const GeneralWalletProvider: React.FC<{
 		}
 	};
 
+	const isOnSolana = walletChainType === ChainType.SOLANA;
+	const isOnEVM = walletChainType === ChainType.EVM;
```

Comment:
Wouldn't it be more efficient if we use useMemo? I doubt that it will improve the efficiency as it's a primitive boolean variable and updating with the same value may not cause and rerender. Just wanted to get your attention and thoughts on this.